daml-lf/transaction/src/main/scala/com/digitalasset/daml/lf/transaction/Node.scala

```diff
@@ -205,12 +209,14 @@ object Node {
             signatories2,
             controllers2,
             _,
-            exerciseResult2) =>
+            exerciseResult2,
+            key2) =>
           import ne._
           targetCoid === targetCoid2 && templateId == templateId2 && choiceId == choiceId2 &&
           consuming == consuming2 && actingParties == actingParties2 && chosenValue === chosenValue2 &&
           stakeholders == stakeholders2 && signatories == signatories2 && controllers == controllers2 &&
-          exerciseResult.fold(true)(_ => exerciseResult === exerciseResult2)
+          exerciseResult.fold(true)(_ => exerciseResult === exerciseResult2) &&
+          key === key2
```

Comment:
If validating a past transaction (where key = None) will produce a NodeExercises where key = Some, then this needs to use the same approach as `exerciseResult`.

Reply:
thanks, good catch!